sdk/java/runtime/main.go

```
@@ -298,23 +298,28 @@ func (m *JavaSdk) finalJar(
 func (m *JavaSdk) mvnContainer(ctx context.Context) *dagger.Container {
 	ctr := dag.
 		Container().
-		From(fmt.Sprintf("%s@%s", MavenImage, MavenDigest))
-	if platform, err := ctr.Platform(ctx); err == nil && strings.Contains(string(platform), "arm64") {
-		ctr = ctr.WithEnvVariable("MAVEN_OPTS", "-XX:UseSVE=0")
-	}
+		From(fmt.Sprintf("%s@%s", MavenImage, MavenDigest)).
+		With(disableSVEOnArm64(ctx))
 	return ctr
 }
 
 func (m *JavaSdk) jreContainer(ctx context.Context) *dagger.Container {
 	ctr := dag.
 		Container().
-		From(fmt.Sprintf("%s@%s", JavaImage, JavaDigest))
-	if platform, err := ctr.Platform(ctx); err == nil && strings.Contains(string(platform), "arm64") {
-		ctr = ctr.WithEnvVariable("JAVA_OPTS", "-XX:UseSVE=0")
-	}
+		From(fmt.Sprintf("%s@%s", JavaImage, JavaDigest)).
+		With(disableSVEOnArm64(ctx))
 	return ctr
 }
 
+func disableSVEOnArm64(ctx context.Context) dagger.WithContainerFunc {
+	return func(ctr *dagger.Container) *dagger.Container {
+		if platform, err := ctr.Platform(ctx); err == nil && strings.Contains(string(platform), "arm64") {
```

`err` could be a lot of things here, ideally we shouldn't silently fail.